src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,51 @@
+.calendar {
+  position: relative;
+  display: flex;
+  flex-wrap: wrap;
+  width: $calendar-width;
+  gap: $gap;
+  margin: 25vh auto;
+
+  @each $month in $month-lengths {
+    &--month-length--#{$month} {
+      .calendar__day:nth-child(n + #{$month + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    @each $day, $index in $days-week {
+      &--first__day-#{$day} {
```

Comment:
The class name should be '--start-day-#{$day}' instead of '--first__day-#{$day}' to follow BEM naming conventions and match the task description.